src/cpp/library/exception

```diff
@@ -70,6 +77,9 @@ public:
 
   exception& operator=(const exception&) throw ();
 
+  /**
+   * WARNING: throw() is an alias for noexcept(true) in C++17, but bad for C++14
```

Comment:
Why is it "bad" for C++14? Should we do something about that by, e.g., conditional to `__cplusplus` disabling it?